core/tee/tee_svc_cryp.c

```diff
@@ -3409,47 +3474,75 @@ static TEE_Result get_sm2_kep_params(const TEE_Attribute *params,
 			found |= BIT(IS_INITIATOR);
 			break;
 		case TEE_ATTR_ECC_PUBLIC_VALUE_X:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
-					     p->content.ref.length,
+			if (memdup_user(p->content.ref.buffer,
+					p->content.ref.length, &bufs[n]))
+				return TEE_ERROR_OUT_OF_MEMORY;
```

Comment:
return the error code returned by `memdup_user()`

ditto at lines 3488, 3498, 3508 etc...